src/v/cluster/controller_backend.cc

```diff
@@ -305,10 +305,25 @@ controller_backend::bootstrap_ntp(const model::ntp& ntp, deltas_t& deltas) {
           ntp);
         // if we found update finished operation it is preceeding operation that
         // created partition on current node
-        if (it->type == op_t::update_finished) {
+        /**
+         * At this point we may have two situations
+         * 1. replica was created on current node shard when partition was
+         *    created, with addition delta, in this case `it` contains this
+         *    addition delta.
+         *
+         * 2. replica was moved to this node with `update` delta type, in this
+         *    case `it` contains either `update_finished` delta from previous
+         *    operation or `add` delta from previous operation. In this case
+         *    operation does not contain current node and we need to execute
+         *    operation that is following the found one as this is the first
+         *    operation that created replica on current node
+         *
+         */
+        if (!contains_node(_self, it->new_assignment.replicas)) {
```

Comment:
it's hard to review the solution because it's not clear what the root cause problem is. the commit message only states that "Previously it might happen that the revision was incorrectly determined", not how the system arrives at the state where an incorrect revision was determined.

Reply:
i've added a message describing the issue.

Reply:
thanks!